### PR TITLE
Remove irrelevant flags in all browsers for Headers API

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR removes irrelevant flag data for all browsers for the `Headers` API as per the corresponding [data guidelines](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines.md#removal-of-irrelevant-flag-data).

This PR was created from results of a [script](https://github.com/vinyldarkscratch/browser-compat-data/blob/scripts/remove-redundant-flags/scripts/remove-redundant-flags.js) designed to remove irrelevant flags.

Paired with #9916 and #9917.

### CODE REVIEW

File: api/Headers.json [hunk 1]
```diff
@@ -4,85 +4,30 @@
...
-              "flags": [
-                {
-                  "type": "preference",
-                  "name": "dom.fetch.enabled"
-                }
-              ]
-            }
-          ],
+          "firefox": {
+            "version_added": "39"
```
Comment (foolip) [inline api/Headers.json:17]: I doubt this data is correct, that `fetch()` would have shipped earlier in desktop than on Android. This change makes the difference more obvious. Are you able to test for when this was shipped in Firefox for Android?

Looks like the data came from wiki migration: https://github.com/mdn/browser-compat-data/pull/929

File: api/Headers.json [hunk 2]
```diff
@@ -107,85 +52,30 @@
...
-                    "name": "dom.fetch.enabled"
-                  }
-                ]
-              }
-            ],
+            "firefox": {
+              "version_added": "39"
+            },
             "firefox_android": {
               "version_added": false
```
Comment (foolip) [inline api/Headers.json:68]: This seems implausible.

File: api/Headers.json [hunk 3]
```diff
@@ -854,7 +515,7 @@
...
             },
             "opera": {
-              "version_added": "28"
+              "version_added": "29"
```
Comment (foolip) [inline api/Headers.json:518]: Right, Opera 29 is based on Chromium 42, this seems to have been wrong.

**Review by foolip — APPROVED** [review]
There seems to be a lot wrong with the Firefox for Android data here. I don't think it should be fixed in this PR, but it could be clearer if it's fixed before this PR. I'm approving since either way would be fine, but not merging since I'm not sure.

File: api/Headers.json (hunk 1 above)
Comment (vinyldarkscratch) [reply api/Headers.json:17]: I'll have to see if I can run a Firefox Android instance that old, but for now, I've opened up #9880.

**Comment by ddbeck** [comment]
> I'm approving since either way would be fine, but not merging since I'm not sure.

@vinyldarkscratch @foolip what's the plan here? Are changes planned or should this get merged?

**Comment by foolip** [comment]
Over to @vinyldarkscratch to decide on the order.

**Comment by vinyldarkscratch** [comment]
I'm not quite ready to test old Firefox Android versions (there's more steps I need to perform to get my test environment ready), so I'd like to get this PR merged first if that's alright!

**Pull request merged**